src/redux/reducers/fetchApps.reducer.js

```diff
@@ -0,0 +1,38 @@
+const SET_APPS = "SET_APPS"
+const SET_IS_LOADING = "SET_IS_LOADING"
+
+export const INITIAL_STATE = {
+  allApps: [],
+  isLoading: true
+};
+    
+// TODO: Write the reducer cases for API list
+export const fetchAppsReducer = (state = INITIAL_STATE, action = {}) => {
+  switch (action.type) {
+    case SET_IS_LOADING:
+      return {
+        ...state,
+        isLoading: action.payload
+      };
+    case SET_APPS:
+      return {
+        ...state,
+        allApps: action.payload
+      };
+    default: {
+      return state;
+    }
+  }
+};
+
```

Comment:
if we are not using useEffect do you still want to keep these lines? I suggest you can remove it if we dont use it please